### ISSUE TITLE
[Bug] RMSNorm parameters stay at 1 and do not update

### ISSUE THREAD

**Issue by pprp** [fla-org/flame#47]
### Describe the Bug

After pretraining, the weight of RMSNorm remain all 1 and seems not get updated. 

<img width="1180" height="462" alt="Image" src="https://github.com/user-attachments/assets/c14b74e1-5544-4beb-81db-44adf8f40fb8" />

### Steps to Reproduce the Bug

````
#!/bin/bash
set -x
set -euo pipefail

if [[ $# -gt 0 && ! "$1" =~ ^-- ]]; then
    NOTE="$1"
    shift
else
    NOTE=""
fi

CUSTOM_ARGS=("$@")

export MASTER_PORT=${MASTER_PORT:-$(shuf -i 20000-65000 -n 1)}
export NNODE=${NNODE:-1}

DEFAULT_LAMODEL_DIR=${DEFAULT_LAMODEL_DIR:-"path/to/model"}
TRAINING_DATASET=${TRAINING_DATASET:-"path/to/dataset"}
SEQLEN=${SEQLEN:-65536}
TRAIN_STEPS=${TRAIN_STEPS:-20480}
MODEL_SETTING=${MODEL_SETTING:-gla_1.3B}
TOKENIZER_SETTING=${TOKENIZER_SETTING:-gla-1.3B-100B}
DTYPE=${DTYPE:-bf16}

wandb offline

case "$DTYPE" in
  bf16)
    LOG_DIR=./logs/${MODEL_SETTING}_logs_bf16_${NOTE}
    DUMP_PREFIX=exp/${MODEL_SETTING}_exp_bf16_${NOTE}
    EXTRA_ARGS=()
    ;;
  fp8_tensorwise)
    LOG_DIR=./logs/${MODEL_SETTING}_logs_tensorwise_fp8_${NOTE}
    DUMP_PREFIX=exp/${MODEL_SETTING}_exp_tw_fp8_${NOTE}
    EXTRA_ARGS=(--float8.recipe_name tensorwise)
    ;;
  *)
    echo "Unknown DTYPE: $DTYPE"; exit 1;;
esac

mkdir -p "$LOG_DIR"

MODEL_TAG=$(echo "$MODEL_SETTING" | tr '_' '-')
if [ "$DTYPE" = "bf16" ]; then
  DUMP_DIR="$DUMP_PREFIX/${MODEL_SETTING}/${MODEL_SETTING}-4K-100B/batch1.seqlen65536.context4096.warmup1024.update1.steps${TRAIN_STEPS}.lr1e-3.cosine"
else
  DUMP_DIR="$DUMP_PREFIX/${MODEL_TAG}-4K-100B/batch1.seqlen65536.context4096.warmup1024.update1.steps${TRAIN_STEPS}.lr1e-3.cosine"
fi
mkdir -p "$(dirname "$DUMP_DIR")"

LOG_FILE="$LOG_DIR/$(date +%Y-%m-%d-%H%M%S).log"

echo "🚀 Running with default config + custom arguments: ${CUSTOM_ARGS[*]}"
bash scripts/train.sh \
  --job.config_file flame/models/fla.toml \
  --job.dump_folder "${DUMP_DIR}" \
  --model.config "configs/${MODEL_SETTING}.json" \
  --model.tokenizer_path "${DEFAULT_LAMODEL_DIR}/${TOKENIZER_SETTING}" \
  --optimizer.name AdamW \
  --optimizer.eps 1e-15 \
  --optimizer.lr 1e-3 \
  --lr_scheduler.warmup_steps 1024 \
  --lr_scheduler.lr_min 0.1 \
  --lr_scheduler.decay_type cosine \
  --training.batch_size 1 \
  --training.seq_len ${SEQLEN} \
  --training.context_len 4096 \
  --training.varlen \
  --training.gradient_accumulation_steps 1 \
  --training.steps ${TRAIN_STEPS} \
  --training.max_norm 1.0 \
  --training.skip_nan_inf \
  --training.dataset ${TRAINING_DATASET} \
  --training.dataset_name default \
  --training.dataset_split train \
  --training.num_workers 32 \
  --training.prefetch_factor 2 \
  --training.seed 42 \
  --checkpoint.interval 2048 \
  --checkpoint.load_step -1 \
  --checkpoint.keep_latest_k 2 \
  --metrics.enable_tensorboard \
  --metrics.log_freq 1 \
  "${EXTRA_ARGS[@]}" \
  "${CUSTOM_ARGS[@]}" 2>&1 | tee -a "$LOG_FILE"
```

### Expected Behavior

We expected the weight of norm is not all 1 

### Environment Information

pytorch-triton             3.3.1+gitc8757738
torch                      2.8.0

**Comment by pprp** [comment]
I have tested torchtitan v0.1.0 and find the weight of RMSNorm can update normally.

**Comment by pprp** [comment]
After discussing with Yu Zhang (@yzhangcs), we identified the main issue: the master weights remain in bfloat16 during the RMSNorm updates.

Since the gradients are around 1e‑3 to 1e‑4 and the learning rate is 1e‑3, bfloat16 does not provide sufficient precision for meaningful updates. As a result, the RMSNorm weights effectively stay fixed at 1.
